lib/cuffsert/yaml-ext.rb

```diff
@@ -0,0 +1,22 @@
+require 'yaml'
+
+module YAML
+  %w[Ref].each do |name|
+    add_domain_type('cuffsert', name) do |tag, value|
+      {name => value}
+    end
+  end
+
+  add_domain_type('cuffsert', 'GetAtt') do |_, value|
+    {'Fn::GetAtt' => value.to_s.split('.')}
+  end
+
+  %w[
+    Base64 Cidr FindInMap GetAZs ImportValue Join Select Split Sub Transform
+    And Equals If Not Or
+  ].each do |name|
+    add_domain_type('cuffsert', name) do |tag, value|
+      {['Fn', name].join('::') => value}
```

Comment:
Does this work with for example

```yaml
Name: !Sub
  - www.${Domain}
  - { Domain: !Ref RootDomainName }
```

(example from https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html)

i.e. is `value` the list in that case?

Reply:
Yes, Psych expands nested tags. Added example to illustrate this.